lib/src/markdown_content.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:markdown/markdown.dart';
+import 'package:meta/meta.dart';
+import 'package:html/parser.dart' as html_parser;
+import 'package:path/path.dart' as p;
+
+import 'model.dart';
+
+/// The extracted content of a markdown file.
+class ScannedMarkdownContent {
+  String parseError;
+  List<String> images;
+  List<String> links;
+
+  Map<String, dynamic> toJson() => <String, dynamic>{
+        'parseError': parseError,
+        'images': images,
+        'links': links,
+      };
+}
+
+/// Scans a markdown text and extracts its content.
+ScannedMarkdownContent scanMarkdownText(String text) {
+  try {
+    final htmlText = markdownToHtml(text);
+    final html = html_parser.parseFragment(htmlText);
+    final images = html
+        .querySelectorAll('img')
+        .where((e) => e.attributes.containsKey('src'))
+        .map((e) => e.attributes['src'])
+        .toSet()
+        .toList();
+    images.sort();
+    final links = html
+        .querySelectorAll('a')
+        .where((e) => e.attributes.containsKey('href'))
+        .map((e) => e.attributes['href'])
+        .toSet()
+        .toList();
+    links.sort();
+    return ScannedMarkdownContent()
+      ..images = images
+      ..links = links;
+  } catch (e) {
+    return ScannedMarkdownContent()..parseError = e.toString();
+  }
+}
+
+/// Scans a markdown file and extracts its content.
+Future<ScannedMarkdownContent> scanMarkdownFileContent(File file) async {
+  final text = await file.readAsString();
+  return scanMarkdownText(text);
+}
+
+/// Analyze a markdown file and return a composite suggestion.
+Future<Suggestion> analyzeMarkdownFile(File file, {String pkgDir}) async {
```

Comment:
I've started with that, but wanted to keep it simple for now (e.g. it would introduce new suggestion codes, much longer code to handle it).
If -later- we were also checking whether the URLs exist, it probably could be a list.

However, the location of the link will probably be not reported for now, as the extraction of the information would require more complex markdown AST processing.

Reply:
At least reporting the link text itself might help in determining what is wrong.

Reply:
I'd keep it simple for now, we can always add more info to it later.